src/app/shared/services/action-state.service.ts

```diff
@@ -29,35 +30,35 @@ export class ActionStateService {
   }
 
   private onReconnected(): void {
-    this.actionsSubjects.forEach((_, rundownId: string) => this.resetActionsSubject(rundownId))
+    this.actionsSubjects.forEach((_, rundownId?: string) => this.resetActionsSubject(rundownId))
```

Comment:
Why is `rundownId` made optional?

Reply:
From reading a bit more into the code, my guess is that it is made optional, to be able to call the non-rundown specific GET actions endpoint.
I don't think the optionality of rundownIds should be there. I think for the `ActionStateService` the best approach would be to rename `resetActionsSubject` to `resetActionsSubjectForRundown` and then create a new one method:
```typescript
private resetSystemActionsSubject(): void {
  const systemActionsSubject: BehaviourSubject<Action[]> | undefined = this.getActionsSubject(SYSTEM_ACTIONS_ID)
  if (!actionSubject) {
    return
  }
this.logger.debug(`Resetting system actions`)
this.fetchSystemActions()
  .then(systemActions => systemActionsSubject.next(systemActions))
  .catch(error => this.logger.data(error).error('Encountered an error while fetching system actions.')
}
```

The same approach could be used for subscribing to rundown actions and system actions.
I know that there will be some duplication, but is is less invasive than the the optionality of rundown ids.

Reply:
I'm not too sure exactly what the proposal is, I'll wait until you have time to talk, so you can explain the idea in a little more detail.

Reply:
I am happy with the solution we made while talking. So I'll resolve this thread.

Reply:
Done